File: 카카오/추석트래픽.py
def solution(lines):
    answer = 0
    start_time = []
    end_time = []
    def get_time(t):
        h,m,s = t.split(":");
        h = int(h)*3600
        m = int(m)*60
        s,mil = s.split(".")
        s = int(s)
        mil = int(mil)
        return (h+m+s)*1000+mil
    
    def get_start_time(t,millsecond):
        mil = millsecond[:-1]
        mil = int(float(mil)*1000)
        return get_time(t)-mil+1
    
    for i in lines:
        time = i.split()
        start_time.append(get_start_time(time[1],time[2]))
        end_time.append(get_time(time[1]))
    
    for i in range(len(lines)):
        cnt = 0
        cur_time = end_time[i]
        for j in range(i,len(lines)):
            if cur_time>start_time[j]-1000:
                cnt+=1
        answer = max(answer,cnt)
    return answer

File: 프로그래머스/추석트래픽.js
function solution(lines) {
  var answer = 0;
  let start_time = [];
  let end_time = [];
  function gettime(time) {
    let [hour, minute, sec] = time.split(":");
    hour = Number(hour) * 3600;
    minute = Number(minute) * 60;
    sec = Number(sec);
    return hour + minute + sec;
  }
  function get_start(time, load) {
    return gettime(time) - Number(load.slice(0, load.length - 1)) + 0.001;
  }
  for (const i of lines) {
    const [a, b, c] = i.split(" ");
    start_time.push(get_start(b, c));
    end_time.push(gettime(b));
  }

  for (let i = 0; i < lines.length; i++) {
    let count = 1;
    let cur = end_time[i];
    for (let j = i + 1; j < lines.length; j++) {
      if (end_time[i] > start_time[j] - 1) {
        count++;
      }
    }
    answer = Math.max(count, answer);
  }
  return answer;
}
